File: Day 2/day2.ts
function createFrame(names: string[]): string {
    const maxLen = Math.max(...names.map(name => name.length)) + 4;

    let res = '*'.repeat(maxLen) + "\n";
    for (const name of names){
        const paddedName = `* ${name}`;
        const spaces = ' '.repeat(maxLen - 1 - paddedName.length);
        res += paddedName + spaces + "\n";
    }

    res += '*'.repeat(maxLen);
    return res;
}

File: Day 2/day2.py
def create_frame(names: list[str]) -> str:
    max_len = max([len(name) for name in names]) + 4

    res = '*'*max_len + "\n"
    for name in names:
        name = '* ' + name
        spaces = ' '*(max_len-1 - len(name)) 
        res += name + spaces + '*\n'

    res += '*'*max_len
    return res